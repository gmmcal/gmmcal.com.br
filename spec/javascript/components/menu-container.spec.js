import { mount, shallowMount } from '@vue/test-utils'
import MenuContainer from '@/components/menu-container'
import Menu from '@/components/menu'
import create from '../settings/utils'

describe('MenuContainer', () => {
  let wrapper

  beforeEach(() => {
    window.gon = {
      menu_links: create('menu_items')
    }

    wrapper = mount(MenuContainer)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('has data properties', () => {
    const expected = ['links', 'click']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('Child components', () => {
    it('includes Menu', () => {
      const child_component = wrapper.find(Menu)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    it('renders 1 Menu objects', () => {
      const props = {
        propsData: {
          items: gon.menu_links
        }
      }
      wrapper = shallowMount(Menu, props)
      const menus = wrapper.findAll(Menu)
      expect(menus.length).toBe(1)
    })
  })
})
