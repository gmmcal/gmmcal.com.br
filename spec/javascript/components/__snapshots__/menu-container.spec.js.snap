// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuContainer renders correctly 1`] = `
<div class="navbar navbar-expand-md navbar-dark"><button aria-label="Menu" type="button" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div class="menu-container collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="active"><a data-turbolinks="false" href="#home" alt="home" class="home">Home</a></li>
      <li><a data-turbolinks="false" href="#about" alt="about" class="about">About</a></li>
    </ul>
  </div>
</div>
`;
